### PR TITLE
Add Feature Form View Tutorial

### PR DESCRIPTION
**Pull request by mhdostal**
Resolves Apollo #129: Swift - Toolkit - Tutorial & Documentation

Adds the files necessary for the FeatureFormView Tutorial.

### CODE REVIEW

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep1.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import SwiftUI
+import ArcGIS
+import ArcGISToolkit
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep1.swift:3]:
```suggestion
import ArcGIS
import ArcGISToolkit
import SwiftUI
```

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep3.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import SwiftUI
+import ArcGIS
+import ArcGISToolkit
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep3.swift:3]:
```suggestion
import ArcGIS
import ArcGISToolkit
import SwiftUI
```

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep4.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import SwiftUI
+import ArcGIS
+import ArcGISToolkit
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep4.swift:3]:
```suggestion
import ArcGIS
import ArcGISToolkit
import SwiftUI
```

**Review by dfeinzimer — COMMENTED** [review]
Looks good so far.

This should probably also delete "Coming Soon" from [this line](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/d58c7665068b357d9c48a0ce080ca00c4aa548c4/Sources/ArcGISToolkit/Components/FeatureFormView/FeatureFormView.swift#L56).

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    @State private var featureForm: FeatureForm? {
+        didSet { showFeatureForm = featureForm != nil }
+    }
+    
+    @State private var showFeatureForm = false
+    
+    @State private var floatingPanelDetent: FloatingPanelDetent = .full
+    
+    /// A Boolean value indicating whether the alert confirming the user's intent to cancel is displayed.
+    @State private var isCancelConfirmationPresented = false
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift:29]: Should we call this `cancelConfirmationIsPresented` to fulfill the "reads like an assertion" guideline? I can update the example view separately if you like.

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep2.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    
+    @StateObject private var dataModel = MapDataModel(
+        map: makeMap()
+    )
+    
+    @State private var identifyScreenPoint: CGPoint?
+    
+    @State private var featureForm: FeatureForm?
+    
+    @State private var showFeatureForm = false
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep2.swift:22]: Just a suggestion to match the "reads like an assertion" guideline. Take or leave.

```suggestion
    @State private var featureFormIsPresented = false
```

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+                            }
+                        }
+                    }
+                }
+        }
+    }
+    
+    /// Submit the changes made to the form.
+    func submitChanges() async {
+        guard let featureForm = featureForm,
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift:100]:
```suggestion
        guard let featureForm,
```

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    
+    @State private var featureForm: FeatureForm? {
+        didSet { showFeatureForm = featureForm != nil }
+    }
+    
+    @State private var showFeatureForm = false
+    
+    @State private var floatingPanelDetent: FloatingPanelDetent = .full
+    
+    /// A Boolean value indicating whether the alert confirming the user's intent to cancel is displayed.
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift:28]: Looks like this is the only doc comment that made it in.

```suggestion
```

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift (hunk 6 above)
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift:100]: Fixing in example in #653.

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep2.swift (hunk 5 above)
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep2.swift:22]: Fixing in example in #653.

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep1.swift [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+import SwiftUI
+
+struct FeatureFormExampleView: View {
+    static func makeMap() -> Map {
+        let portalItem = PortalItem(
+            portal: .arcGISOnline(connection: .anonymous),
+            id: Item.ID("9f3a674e998f461580006e626611f9ad")!
```
[the remark this answers is not in the document]
Comment (mhdostal) [reply Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep1.swift:9]: Mark B is creating a map we can use, so we'll just need to update this with that when it's ready.

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep1.swift [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+        )
+        return Map(item: portalItem)
+    }
+    
+    @StateObject private var dataModel = MapDataModel(
```
[the remark this answers is not in the document]
Comment (mhdostal) [reply Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep1.swift:14]: Ok, if you find it, we can change this.

[the remark this answers is not in the document]
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep1.swift:14]: Found it with the help of Phil R! [Here is the relevant doc](https://developers.arcgis.com/swift/api-reference/documentation/arcgis/workingwithgeoviews). 

There's also a good description in the PR that changed that doc. See Swift 3698.
Overall it seems that `@State` is sufficient but `@StateObject` is also okay so it's your call.

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep1.swift (hunk 8 above)
[the remark this answers is not in the document]
Comment (mhdostal) [reply Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep1.swift:9]: Ok, thanks. I made the changes to a `@State` property, as that was simpler.

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep6.swift [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    @State private var floatingPanelDetent: FloatingPanelDetent = .full
+    
+    var body: some View {
+        MapViewReader { proxy in
+            MapView(map: map)
+                .onSingleTapGesture { screenPoint, _ in
+                    identifyScreenPoint = screenPoint
+                }
+                .task(id: identifyScreenPoint) {
+                    guard let identifyScreenPoint else { return nil }
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep6.swift:33]:
```suggestion
                    guard let identifyScreenPoint else { return }
```

File: Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+    @State private var cancelConfirmationIsPresented = false
+    
+    var body: some View {
+        MapViewReader { proxy in
+            MapView(map: map)
+                .onSingleTapGesture { screenPoint, _ in
+                    identifyScreenPoint = screenPoint
+                }
+                .task(id: identifyScreenPoint) {
+                    guard let identifyScreenPoint else { return nil }
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Documentation.docc/Resources/FeatureFormView/FeatureFormViewStep7.swift:35]:
```suggestion
                    guard let identifyScreenPoint else { return }
```

**Review by dfeinzimer — COMMENTED** [review]
Just the last two `nil`s and then I think this is set.

**Comment by mhdostal** [comment]
Thanks @dfeinzimer !

**Pull request merged**